src/App/Pages/Vault/ScanPage.xaml.cs

```diff
@@ -26,20 +28,17 @@ public partial class ScanPage : BaseContentPage
         private bool _pageIsActive;
         private bool _qrcodeFound;
         private float _scale;
-
+        private ZXingScannerView _zxing;
         private readonly LazyResolve<ILogger> _logger = new LazyResolve<ILogger>("logger");
 
-        public ScanPage(Action<string> callback)
+        public ScanPage(Action<string> callback, bool hasCameraPermission)
```

Comment:
I don't think the caller should have the responsibility of telling this page whether the app has camera permission. IMO this Page/ViewModel should be the responsible of checking whether the camera is allowed, so I'd remove this parameter from the ctor and check the permission on the spot that is needed.